synapse/storage/data_stores/main/state.py

```diff
@@ -60,24 +60,31 @@ class StateGroupWorkerStore(EventsWorkerStore, SQLBaseStore):
     def __init__(self, database: Database, db_conn, hs):
         super(StateGroupWorkerStore, self).__init__(database, db_conn, hs)
 
-    @defer.inlineCallbacks
-    def get_room_version(self, room_id):
+    @cached(max_entries=10000)
```

Comment:
ohhh did I miss the memo where async functions can be cached?

Reply:
Oh, yes, turns out that somewhere in the depths the cache descriptor calls `run_in_background` on the cahced function